### PR TITLE
no-os submodule sync + projects updated

### PR DESCRIPTION
**Pull request by amiclaus**
Sync the ADICUP3029 projects with the latest changes on the no-OS
repository.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Review by amiclaus — COMMENTED** [review]
V2:
- sync no-OS submodule, bringing the ad7799_iio changes required for the cn0548 project.

File: projects/ADuCM3029_demo_cn0567/src.mk [hunk 1]
```diff
@@ -4,9 +4,9 @@ SRC_DIRS += $(NO-OS)/iio/iio_app
 SRC_DIRS += $(PLATFORM_DRIVERS)
 SRC_DIRS += $(INCLUDE)
 SRC_DIRS += $(DRIVERS)/photo-electronic/adpd410x
-SRC_DIRS += $(NO-OS)/drivers/gpio
-SRC_DIRS += $(NO-OS)/drivers/spi
-SRC_DIRS += $(NO-OS)/drivers/irq
+SRC_DIRS += $(NO-OS)/drivers/api
+SRC_DIRS += $(NO-OS)/drivers/api
+SRC_DIRS += $(NO-OS)/drivers/api
```
Comment (CiprianRegus) [inline projects/ADuCM3029_demo_cn0567/src.mk:9]: These lines seem to include the same directory, is this intentional?

**Review by amiclaus — COMMENTED** [review]
> These lines seem to include the same directory, is this intentional?

Good catch! Will fix.

**Review by amiclaus — COMMENTED** [review]
V3:
 - remove duplicated `drivers/api` includes in src.mk

**Review by CiprianRegus — APPROVED** [review]
Looks good.

**Pull request merged**